### PR TITLE
Add agentPoolVersionProfile Object Property

### PR DESCRIPTION
**Pull request by gtxistxgao**
Add the new field in agent pool "properties":

Another new property "**AgentPoolVersionProfile**" will be introduced. It will contains a new property called "nodeImageVersion". Use "AgentPoolVersionProfile" instead of directly put "nodeImageVersion" to enable us adding more version info into this profile.

```
{
  "properties": {
    "count": 2,
    "vmSize": "Standard_DS3_v2",
    "osDiskSizeGB": 30,
    "maxPods": 110,
    "type": "VirtualMachineScaleSets",
    "orchestratorVersion": "1.13.12",
    "osType": "Linux",
    "agentPoolVersionProfile": {
      "nodeImageVersion": "latest"
    }
  }
}
```

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/examples/AgentPoolsCreate_Update.json [hunk 1]
```diff
@@ -20,8 +20,9 @@
         "nodeTaints": [
           "Key1=Value1:NoSchedule"
         ],
-        "scaleSetPriority": "Low",
-        "scaleSetEvictionPolicy": "Delete"
```
Comment (gtxistxgao) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/examples/AgentPoolsCreate_Update.json]: those a not needed. removed it here.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json [hunk 2]
```diff
@@ -1522,6 +1522,11 @@
             "type": "string"
           },
           "description": "Taints added to new nodes during node pool create and scale. For example, key=value:NoSchedule."
+        },
+        "agentPoolVersionProfile": {
```
Comment (ryansbenson) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json]: This constitutes a breaking change and requires a new API version.

See 'New property added to response'
https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/guestaccess.aspx?e=xEcppo&share=EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw

**Pull request closed without merging**